cmake/Packaging.cmake

```diff
@@ -139,6 +138,48 @@ ly_install(FILES ${_cmake_package_dest}
     COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
 )
 
+# Do inital check on workspace path in the event that LY_3RDPARTY_PATH is misconfigured
+file(REAL_PATH "${CMAKE_CURRENT_SOURCE_DIR}/.." _root_path)
+if (EXISTS "${_root_path}/3rdParty/packages")
+    set(CPACK_LY_3P_PACKAGE_DIRECTORY "${_root_path}/3rdParty/packages")
+else()
+    set(CPACK_LY_3P_PACKAGE_DIRECTORY ${LY_3RDPARTY_PATH}/packages)
+endif()
+
+# Scan the engine and 3rd Party folders for licenses
+file(TO_NATIVE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/python/python.cmd" _python_cmd)
+file(TO_NATIVE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/scripts/license_scanner" _license_script_path)
+
+set(_license_script "${_license_script_path}/license_scanner.py")
+set(_license_config "${_license_script_path}/scanner_config.json")
+
+set(_license_scan_path "${CMAKE_CURRENT_SOURCE_DIR}" "${CPACK_LY_3P_PACKAGE_DIRECTORY}")
+set(CPACK_3P_LICENSE_FILE "${CPACK_BINARY_DIR}/NOTICES.txt")
+set(CPACK_3P_MANIFEST_FILE "${CPACK_BINARY_DIR}/SPDX-License.json")
+
+set(_license_command
+    ${_python_cmd} -s
+    -u ${_license_script}
+    --config-file ${_license_config}
+    --license-file-path ${CPACK_3P_LICENSE_FILE}
+    --package-file-path ${CPACK_3P_MANIFEST_FILE}
+)
+
+message(STATUS "Scanning for license files in ${_license_scan_path}")
+execute_process(
+    COMMAND ${_license_command} --scan-path ${_license_scan_path}
+    RESULT_VARIABLE _license_result
+    ERROR_VARIABLE _license_errors
+    OUTPUT_VARIABLE _license_output
+    ECHO_OUTPUT_VARIABLE
+)
```

Comment:
the execution of this scrip tshould be done as part of packaging/install and not part of configuration
This will execute the license scan every time a developer generates. Meaning, this will execute when the user does `cmake ...` and not when the user does `cpack ...`
To inject into the packaging step, you can do the same we do at the end of the file:
```
file(APPEND "${CPACK_OUTPUT_CONFIG_FILE}" "
   # code that goes into packaging execution
")
```
Or, you can move this logic to `Platform/Windows/PackagingPreBuild.cmake` (I renamed this to `Platform/Windows/PackagingPreBuild_windows.cmake`)
If you want to create common code across platforms, you can create a `Platform/Common/PackagingPreBuild_common.cmake` and include it from `Platform/Windows/PackagingPreBuild.cmake` (I created a common file for PostBuild)

Reply:
I actually had that in the PreBuild, but seemes like I misunderstood where it should live. I'll move it back

Reply:
yeah, is really not straightforward, there are several stages in play here:
- configuration time, this is execution of the cmake files that are included directly/indirectly by the root CMakeLIsts.txt
- generation time, this is a step after configuration time where it generates the build scripts for a particular build system (e.g. in Windows the vcxproj, sln files). There isn't much you can do here, but generation expressions get expanded here.
- install time. When adding something through [install(](https://cmake.org/cmake/help/latest/command/install.html), that generates into a `cmake_install.cmake` file. When executing the install target or calling cmake install, it executes that `cmake_install.cmake` script.
- cpack time. Similar to install time, cmake generates a `CPackConfig.cmake` file that is executed during cpack. Variables that start with `CPACK_` are passed from configuration time to cpack time. The Packaging[Pre/Post]Build scripts are called from cpack time. Injecting things into cpack is not as easy as it is during install, but you can write directly to the cmake file (`${CPACK_OUTPUT_CONFIG_FILE}`)